flax/training/early_stopping.py

```diff
@@ -45,8 +59,9 @@ def update(self, metric):
     """Update the state based on metric.
     
     Returns:
-      Whether there was an improvement greater than min_delta from
-          the previous best_metric and the updated EarlyStop object.
+      A pair (has_improved, early_stop), where `has_improved` is True when there
```

Comment:
why don't we store the `has_improved` in the dataclass like the `should_stop` ?

would simplify the usage and make it harder to mix up the `has_improved` with something like `should_stop` or some other boolean

Reply:
TBH I was also confused about the API but didn't think about it so carefully. I propose we get this PR in and improve the API in a follow-up